spec/stylelint_spec.rb

```diff
@@ -8,39 +8,68 @@ module Danger
       expect(Danger::DangerStylelint.new(nil)).to be_a Danger::Plugin
     end
 
-    #
-    # You should test your custom attributes and methods here
-    #
     describe "with Dangerfile" do
-      before do
-        @dangerfile = testing_dangerfile
-        @my_plugin = @dangerfile.stylelint
+      let(:stylelint) { testing_dangerfile.stylelint }
+      let(:warnings) { stylelint.status_report[:warnings] }
+      let(:errors) { stylelint.status_report[:errors] }
+      let(:messages) { stylelint.status_report[:messages] }
+      let(:markdowns) { stylelint.status_report[:markdowns] }
 
-        # mock the PR data
-        # you can then use this, eg. github.pr_author, later in the spec
-        json = File.read("#{File.dirname(__FILE__)}/support/fixtures/github_pr.json") # example json: `curl https://api.github.com/repos/danger/danger-plugin-template/pulls/18 > github_pr.json`
-        allow(@my_plugin.github).to receive(:pr_json).and_return(json)
+      before do
+        allow(stylelint.git).to receive(:deleted_files).and_return([])
+        allow(stylelint.git).to receive(:added_files).and_return([])
+        allow(stylelint.git).to receive(:modified_files).and_return([])
+        allow(stylelint.git).to receive(:renamed_files).and_return([])
+        stub_const("Danger::DangerStylelint::DEFAULT_BIN_PATH", "spec/fixtures/stylelint")
       end
 
-      # Some examples for writing tests
-      # You should replace these with your own.
+      it "does not make an empty message" do
+        allow(stylelint).to receive(:lint).and_return("[]")
 
-      it "Warns on a monday" do
-        monday_date = Date.parse("2016-07-11")
-        allow(Date).to receive(:today).and_return monday_date
+        expect(warnings).to eq([])
+        expect(errors).to eq([])
+        expect(messages).to eq([])
+        expect(markdowns).to eq([])
+      end
 
-        @my_plugin.warn_on_mondays
+      it "fails if stylelint not installed" do
+        allow(stylelint).to receive(:bin_path).and_return("")
 
-        expect(@dangerfile.status_report[:warnings]).to eq(["Trying to merge code on a Monday"])
+        expect { stylelint.lint }.to raise_error("stylelint is not installed")
       end
 
-      it "Does nothing on a tuesday" do
-        monday_date = Date.parse("2016-07-12")
-        allow(Date).to receive(:today).and_return monday_date
+      describe "#lint" do
+        let(:valid_result) { JSON.parse(File.read("spec/fixtures/valid.json")) }
+        let(:error_result) { JSON.parse(File.read("spec/fixtures/error.json")) }
+
+        it("lints all files when changes_only is disabled by default") do
```

Comment:
Minor comment

```suggestion
        it "lints all files when changes_only is disabled by default" do
```

Reply:
oh my jest 😅 Thanks Terone for pointing this out 😛 done in de123cae1830f1defd5c96af06cb480faa4d1aec!